packages/expo-cli/src/commands/upload/submission-service/archive-source/ArchiveFileSource.ts

```diff
@@ -0,0 +1,295 @@
+  } else {
+    const tmpPath = pathJoin(os.tmpdir(), pathBasename(url));
+    log('Downloading your app archive');
+    return downloadAppArchiveAsync(url, tmpPath);
+  }
+}
+
+async function getArchiveLocationForPathAsync(mode: SubmissionMode, path: string): Promise<string> {
+  if (mode === SubmissionMode.online) {
+    log('Uploading your app archive to the temporary storage');
+    return await uploadAppArchiveAsync(path);
+  } else {
+    return path;
+  }
+}
+
+async function handleUrlSourceAsync(
+  mode: SubmissionMode,
+  source: ArchiveFileUrlSource
+): Promise<string> {
+  return source.url;
+}
+
+async function handleLatestSourceAsync(
+  mode: SubmissionMode,
+  source: ArchiveFileLatestSource
+): Promise<string> {
+  const { owner, slug } = getAppConfig(source.projectDir);
+  const builds = await StandaloneBuild.getStandaloneBuilds(
+    {
+      platform: source.platform,
+      owner,
+      slug,
+    },
+    1
+  );
+  if (builds.length === 0) {
+    log.error("Couldn't find any builds for this project.");
+    return getArchiveFileLocationAsync(mode, {
+      sourceType: ArchiveFileSourceType.prompt,
+      platform: source.platform,
+      projectDir: source.projectDir,
+    });
+  }
+  return builds[0].artifacts.url;
+}
+
+async function handlePathSourceAsync(
+  mode: SubmissionMode,
+  source: ArchiveFilePathSource
+): Promise<string> {
+  if (!(await existingFile(source.path))) {
+    log.error(`${source.path} doesn't exist`);
+    return getArchiveFileLocationAsync(mode, {
+      sourceType: ArchiveFileSourceType.prompt,
+      platform: source.platform,
+      projectDir: source.projectDir,
+    });
+  }
+  return source.path;
+}
+
+async function handleBuildIdSourceAsync(
+  mode: SubmissionMode,
+  source: ArchiveFileBuildIdSource
+): Promise<string> {
+  const { owner, slug } = getAppConfig(source.projectDir);
+  let build: any;
+  try {
+    build = await StandaloneBuild.getStandaloneBuildById({
+      platform: source.platform,
+      id: source.id,
+      owner,
+      slug,
+    });
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+
+  if (!build) {
+    log.error(`Couldn't find build for id ${source.id}`);
+    return getArchiveFileLocationAsync(mode, {
+      sourceType: ArchiveFileSourceType.prompt,
+      platform: source.platform,
+      projectDir: source.projectDir,
+    });
+  } else {
+    return build.artifacts.url;
+  }
+}
+
+async function handlePromptSourceAsync(
+  mode: SubmissionMode,
+  source: ArchiveFilePromptSource
+): Promise<string> {
+  const { sourceType: sourceTypeRaw } = await prompt({
+    name: 'sourceType',
+    type: 'list',
+    message: 'What would you like to submit?',
+    choices: [
+      { name: 'I have a url to the app archive', value: ArchiveFileSourceType.url },
+      {
+        name: "I'd like to upload the app archive from my computer",
+        value: ArchiveFileSourceType.path,
+      },
+      {
+        name: 'The lastest build from Expo Servers',
```

Comment:
```suggestion
        name: 'The lastest build from Expo servers',
```

Reply:
lower case servers because it's not a proper noun so doesn't require capitalization